### ISSUE TITLE
"UPK" triggers

### ISSUE THREAD

**Issue by jhomer** [Courseplay/courseplay#889]
Not sure if you guys are familiar with this mod but what makes CP understand a certain trigger?
I made a slurry storage mod using the "UniversalProcessKit"
The load and unload triggers don't require an input binding to be activated.
Tried a slurry route from "start" the fill trigger to "waiting point" the unload trigger.
CP drives through the start without stopping (gets filled some as it drives through) but does stop at the waiting point then the "upk" trigger auto unloads and CP drives off again.
What does CP need to see to recognize the fill trigger?

If you guys have time to answer, if not, don't worry about it.
Thanks!

**Comment by ThomasGaertner** [comment]
give us some time....

**Comment by ThomasGaertner** [comment]
can you give us an example to test ? normally it should be implemented already...

**Comment by jhomer** [comment]
Update from previous post: Error on my part, (had filltrigger instead of liquidmanurefilltrigger (type string)) now activation is needed to fill (like default triggers) and now CP stops but does not refill.

Example to test? The mod itself? 1.62MB

**Comment by ThomasGaertner** [comment]
for my understanding: you start a route in mode8 "Manure transporter" , the trailer will be unloaded at the overloading point, but not filled at the fill trigger even if the tractor stops at the fill trigger.
would be good to have your courseplay.xml , the name of your course and the mod for testing reasons, if possible ...

**Comment by jhomer** [comment]
Sent link to download.
Also, my mistake, CP does not stop at the trigger to re-fill in mode 8. When I was testing I also tried mode 4 with a wait point, there it stops but does not refill.

**Comment by JakobTischler** [comment]
At the moment we have no access to the UPK fill triggers. We're in contact with the UPK author and we'll look into implementing support for the fill triggers in the near future.

**Comment by jhomer** [comment]
Here's a link now if you would need to re-download this for testing:
http://fs-uk.com/mods/view/35704/placeable-liquid-manure-storage

**Comment by JakobTischler** [comment]
Alright, I added support for UPK fillTriggers, gasStationTriggers, sprayerFillTriggers etc. But, this will not work until UPK 0.9.3 (or higher) is out, as there are some changes needed in the UPK scripts.

**Comment by jhomer** [comment]
Thank you very much!
I'll test these UPK items when the update is out.
Not sure if you want to close this or if it's still open I'll report back here with test results.

**Comment by JakobTischler** [comment]
Let's leave this open until the new UPK version is out and we can confirm that all is working as it should.

**Comment by jhomer** [comment]
Ok. I see the link and will download. I will test multiple triggers and get back to you. It may not be until 24 hours or so from this message but I will get it done.

**Comment by jhomer** [comment]
I tested the liquid manure storage mod I had sent and works just fine had it fill load and unload with no problems.
I tested a tiptrigger for grain from another mod and had no problems.
Couldn't test a water or fuel fill trigger as CP "wrong trailer for this mode"
I'll have to find or make something to test the sprayerfilltrigger yet.
And if there's a fuel fill trigger that lets a vehicle refill (I have not seen yet)
But the tip and liquid manure triggers cover a fair amount already.

This may be another issue but I had noticed that when CP does stop to re-fuel at a default trigger it only re-fuels till its past the low fuel mark then drives off again.

mor2000 did release the update as well (FYI)
http://www.modhoster.com/mods/universalprocesskit--2
